packages/instrumentation-chromadb/tests/instrumentation.test.ts

```diff
@@ -0,0 +1,380 @@
+
+    // Wait for ChromaDB to spin up
+    await new Promise((resolve) => {
+      setTimeout(resolve, 3000);
+    });
+  });
+
+  before(async () => {
+    provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    instrumentation = new ChromaDBInstrumentation({ traceContent: true });
+    instrumentation.setTracerProvider(provider);
+    instrumentation.manuallyInstrument(chromadb);
+  });
+
+  beforeEach(async function () {
+    contextManager = new AsyncHooksContextManager().enable();
+    context.setGlobalContextManager(contextManager);
+    memoryExporter.reset();
+
+    const embeddingFunction = new chromadb.OpenAIEmbeddingFunction({
+      openai_api_key: process.env.OPENAI_API_KEY ?? "",
+    });
+
+    // Create Collection before each test case
+    collection = await chromaDbClient.getOrCreateCollection({
+      name: "my_collection",
+      embeddingFunction,
+    });
+
+    // Add sample data to the collection before each test case
+    await collection.add({
+      ids: ["uri9", "uri10"],
+      embeddings: [
+        [1.5, 2.9, 3.4],
+        [9.8, 2.3, 2.9],
+      ],
+      metadatas: [{ style: "style1" }, { style: "style2" }],
+      documents: ["doc1000101", "doc288822"],
+    });
+  });
+
+  afterEach(async () => {
+    // Delete Collection after every test completion
+    await chromaDbClient.deleteCollection({ name: "my_collection" });
+    memoryExporter.reset();
+    context.disable();
+  });
+
+  after(() => {
+    // Terminate the Chroma client process after tests
+    if (chromaRun.pid) {
+      // DEBT: chromaRun.kill() is not killing spawned process
+      process.exit(0);
+    }
+  });
+
+  it("should set span attributes for Query", async () => {
+    const input: chromadb.QueryParams = {
+      nResults: 2,
+      queryEmbeddings: [
+        [1.1, 2.3, 3.2],
+        [5.1, 4.3, 2.2],
+      ],
+      where: { style: "style2" },
+    };
+    await collection.query(input);
+
+    const spans = memoryExporter.getFinishedSpans();
+    const attributes = spans[1].attributes;
+
+    // Assert input attributes
+    assert.strictEqual(
+      attributes["db.chroma.query.query_embeddings_count"],
+      JSON.stringify(input.queryEmbeddings?.length),
+    );
+    assert.strictEqual(
+      attributes["db.chroma.query.query_texts_count"],
+      JSON.stringify(input.queryTexts?.length),
+    );
+    assert.strictEqual(
+      attributes["db.chroma.query.n_results"],
+      JSON.stringify(input?.nResults),
+    );
+    assert.strictEqual(
+      attributes["db.chroma.query.where"],
+      JSON.stringify(input?.where),
+    );
+    assert.strictEqual(
+      attributes["db.chroma.query.where_document"],
+      JSON.stringify(input?.whereDocument),
+    );
+    assert.strictEqual(
+      attributes["db.chroma.query.include"],
+      JSON.stringify(input?.include),
+    );
+
+    const events = spans[1].events;
+    assert.strictEqual(events[0].name, "db.chroma.request");
```

Comment:
can you assert on the values of the events?

Reply:
Done